### PR TITLE
Handle all RHEL 7.x releases as RHEL 7 in dir.props

### PR DESCRIPTION
**Pull request by omajid**
Instead of hardcoding the specific RHEL 7.x versions, just handle all RHEL 7 releases. This makes this code more future-proof for future RHEL releases.

This is similar to what is being done in src/managed/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs.

### CODE REVIEW

File: dir.props [hunk 1]
```diff
@@ -373,11 +373,7 @@
   <PropertyGroup Condition="'$(OutputRid)' != ''">
     <PackageTargetRid>$(OutputRid)</PackageTargetRid>
     <PackageTargetRid Condition="'$(OutputRid)' == 'osx.10.11-x64'">osx.10.10-x64</PackageTargetRid>
-    <PackageTargetRid Condition="'$(OutputRid)' == 'rhel.7.0-x64'">rhel.7-x64</PackageTargetRid>
-    <PackageTargetRid Condition="'$(OutputRid)' == 'rhel.7.1-x64'">rhel.7-x64</PackageTargetRid>
-    <PackageTargetRid Condition="'$(OutputRid)' == 'rhel.7.2-x64'">rhel.7-x64</PackageTargetRid>
-    <PackageTargetRid Condition="'$(OutputRid)' == 'rhel.7.3-x64'">rhel.7-x64</PackageTargetRid>
-    <PackageTargetRid Condition="'$(OutputRid)' == 'rhel.7.4-x64'">rhel.7-x64</PackageTargetRid>
+    <PackageTargetRid Condition="'$(OutputRid.StartsWith(&quot;rhel.7.&quot;))' and '$(OutputRid.EndsWith(&quot;-x64&quot;))'">rhel.7-x64</PackageTargetRid>
```
Comment (weshaggard) [inline dir.props]: You can change this to `$(OutputRid.StartsWith('rhel.7.')) and $(OutputRid.EndsWith('-x64'))`

Comment (omajid) [reply dir.props]: Hm. I am sure I tried that and it failed with errors. Let me try again.

Comment (omajid) [reply dir.props]: Just tried it:

    /home/omajid/core-setup/dir.props(376,23): error MSB4092: An unexpected token "rhel" was found at character position 26 in condition "'$(OutputRid.StartsWith('rhel.7.'))' and '$(OutputRid.EndsWith('-x64'))'". [/home/omajid/core-setup/src/pkg/deps/deps.csproj] [/home/omajid/core-setup/build.proj]

Comment (weshaggard) [reply dir.props]: Remove the single quotes around the outside of the property. `'$(OutputRid.StartsWith('rhel.7.'))'` -> `$(OutputRid.StartsWith('rhel.7.'))`

Comment (omajid) [reply dir.props]: Thanks!
